wayland/getting-started.txt

```diff
@@ -0,0 +1,249 @@
+modify the depends files of many packages (including mesa, freetype-harfbuzz,
+pango, cairo, etc.) to not include any Xorg dependencies, but few modifications
+are required beyond that.
+
+First, build wayland and wayland-protocols.
+
+Then, build xkeyboard-config, followed by libxkbcommon with wayland support and
+without x11 support.
+
+Finally, build mesa with the wayland platform and without the x11 platform.
+
+Some packages might assume that, because you are building Linux, you must be
+using GLX. GLX is a graphics implementation provided by mesa, but is only built
+on Xorg systems. Wayland uses EGL and GLES. For some packages, like cairo, you
+might have to specify the gl-backend to be used. 
+
+Launching applications in Wayland requires a little more intervention. GTK will
+default to Wayland usually, but adding the following to your environment is a
+good idea
+
++----------------+-------------------------------------------------------------+
+| GTK            | GDK_BACKEND=wayland                                         |
+| Qt             | QT_QPA_PLATFORM=wayland                                     |
+| firefox        | MOZ_ENABLE_WAYLAND=wayland                                  |
++----------------+-------------------------------------------------------------+
+
+
+As far as finding alternatives to traditionally popular Xorg software like dmenu
+or st, checkout lists like [9] and [16]. 
+
+
+[3.0] Post installation
+________________________________________________________________________________
+
+In environments without seatd/systemd/elogind/consolekit the XDG_RUNTIME_DIR
+variable must be set manually. This should be set before trying to launch a
+wayland session. Add something like this to your user's $HOME/.profile:
+
++------------------------------------------------------------------------------+
+|                                                                              |
+|   mkdir -pm 0700 "${XDG_RUNTIME_DIR:=/tmp/$(id -u)-runtime-dir               |
+|   export XDG_RUNTIME_DIR                                                     |
+|                                                                              |
++------------------------------------------------------------------------------+
+
+Install a compositor. Below is a list:
+
++----------+-------------------------------------------------------------------+
+| dwl      |  A dwm-like compositor using wlroots [17]                         |
+| river    |  A compositor inspired by dwm and bspwm using wlroots [18]        |
+| sway     |  A tiling compositor similar to i3 using wlroots [19]             |
+| hikari   |  A freeBSD window manager inspired by cwm using wlroots [20]      |
+| wayfire  |  A 3D compositor inspired by compiz using wlroots [21]            |
+| wio      |  A Wayland compositor replicating Plan 9's rio using wlroots [22] |
+| velox    |  A dwm inspired compositor using wld and swc [23]                 |
```

Comment:
@dilyn-corner have you experimented with velox extensively? Do qt/gtk apps finaly work?
A year ago when I tried it out, these programs crashed when triggering kontextmenus/popups because this protocol was not yet implemented.
Also I think I have successfully tested alacritty as a terminal. Probably much heavier then the whole compositor stack :v

Reply:
Testing `velox` is happening later tonight! I am uncertain it has progressed any further than when you tried it, however. 

`alacritty` was, is, and will continue to be, bloat. `foot` gang

Reply:
I had a bunch of pages opened as reference material for looking into all this and then lost it. Shame.
AFAIK, `velox` /works/ with `firefox` well enough, if a bit buggy. 

If I ever get the gall and hubris, adding `amdgpu` or `radeon` support to `wld` would be pretty tight...